Create functionality to manage user roles

matborowczyk opened the pull request:
# Description

- On the video the empty view is not aligned correctly, it is fixed on the screenshot, I didn't want to record it once again :) 
- error messages for actions, removal and addition of roles are showed through alerts

<img width="1680" alt="Screenshot 2025-06-19 at 17 28 32" src="https://github.com/user-attachments/assets/76dc30d0-adb5-4477-9aa2-2195ea4e6207" />

<img width="1680" alt="Screenshot 2025-06-19 at 17 30 02" src="https://github.com/user-attachments/assets/23c16e2d-62ed-4835-9398-bbd2f278b1c9" />

https://github.com/user-attachments/assets/6fdbc2cb-c14e-4895-a48b-0abf482cec13



closes #6440 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

LukasStordeur commented on src/Slices/UserManagement/UI/Components/RolesMainColumn/RolesMainColumn.tsx:
```diff
@@ -0,0 +1,52 @@
+  if (roles.isSuccess) {
+    return (
+      <Td dataLabel={words("userManagement.roles")} aria-label="roles-success">
+        {[...new Set(roles.data.map((role) => role.name))].join(", ")}
```
This should be a link/button that collapses/expands the row if clicked

LukasStordeur commented on src/Slices/UserManagement/UI/Components/RolesMainColumn/RolesMainColumn.tsx:
```diff
@@ -0,0 +1,52 @@
+ * @param {UseQueryResult<UserRoleInfo[], Error>} roles - The roles of the user.
+ * @returns {React.FC<Props>} The rendered role top column.
+ */
+export const RolesMainColumn = ({ roles, setAlert }: Props) => {
```
What's a main column ?

matborowczyk replied:
renamed to RolesToggleCell

LukasStordeur commented:
It's almost sad to see a new endpoint, where you can't add/delete multiple roles at once. Especially if this is just an array of strings. Disabling the select while the request is pending, and you have a few roles to assign, is a drag.

LukasStordeur commented:
Could you also show the UI now that it is a button/link that can open/close the rows?

LukasStordeur commented on src/Slices/UserManagement/UI/Page.test.tsx:
```diff
@@ -153,21 +330,90 @@ describe("UserManagementPage", () => {
     });
+  });
+
+  it("should display error alert view when users roles API call fails", async () => {
```
```suggestion
  it("should display error alert view when updating roles fails", async () => {
```

matborowczyk replied:
this suggestion is not adequate to what that test checks

LukasStordeur replied:
the original isn't better imo, it makes me think you need a call both for the user, and the roles to populate the roles column 🤔But based on the content of the test, it might be the case, and if that is, then why isn't it present in the getUser query?

LukasStordeur reviewed (commented):
In the error handling, do we uncheck the selected value that triggered the error response and assert this in the tests?

matborowczyk commented:
> In the error handling, do we uncheck the selected value that triggered the error response and assert this in the tests?

We don't check it until it's success, or to better word it, we get selected roles from the endpoint

matborowczyk commented:
added e2e test scenarios, I will provide screenshot from the pipeline as soon as it will end :) 
Updated the label for no roles to get the span instead of div, and to align the look, what do you think?

![Screenshot 2025-06-30 at 11 01 59](https://github.com/user-attachments/assets/d7cfac4e-c270-47f9-b96a-70c00b8bc7fb)
